engine/src/main/java/org/operaton/bpm/engine/impl/bpmn/parser/BpmnParse.java

```diff
@@ -1700,20 +1700,18 @@ protected CompensateEventDefinition parseThrowCompensateEventDefinition(final El
     final String activityRef = compensateEventDefinitionElement.attribute("activityRef");
     boolean waitForCompletion = TRUE.equals(compensateEventDefinitionElement.attribute("waitForCompletion", TRUE));
 
-    if (activityRef != null) {
+    if (activityRef != null && scopeElement.findActivityAtLevelOfSubprocess(activityRef) == null) {
+      Boolean isTriggeredByEvent = scopeElement.getProperties().get(BpmnProperties.TRIGGERED_BY_EVENT);
+      String type = (String) scopeElement.getProperty(PROPERTYNAME_TYPE);
+      if (Boolean.TRUE == isTriggeredByEvent && "subProcess".equals(type)) {
+        scopeElement = scopeElement.getFlowScope();
+      }
       if (scopeElement.findActivityAtLevelOfSubprocess(activityRef) == null) {
-        Boolean isTriggeredByEvent = scopeElement.getProperties().get(BpmnProperties.TRIGGERED_BY_EVENT);
```

Comment:
`scopeElement.findActivityAtLevelOfSubprocess(activityRef) == null` can't be collapsed here to the surrounding if, because scopeElement might be modified and then can be null after it wasn't before